### ISSUE TITLE
Reveal accessibility

### ISSUE THREAD

**Issue by jaarong** [zurb/foundation-sites#10208]
I'm hoping someone here will help me figure this out, because I'm not really sure what WAI-ARIA standard Foundation conforms to and therefore don't know what the correct answer is.  My assumption is that the button that opens a reveal modal should have the aria-haspopup="dialog" value not aria-haspopup="true" ([according to WAI-ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup).  True would indicate that the popup was a menu.  In WAI-ARIA 1.0, true/false were the only values available but it was only supposed to be applied to menus.  So it seems that either this value should be removed or changed depending on the standard your trying conform to.

We are actively trying to make our template more accessible and I ran across this along with [issue 1006](https://github.com/zurb/foundation-sites/issues/10061) which I see has been fixed.  Can someone let me know if I am correct and the value should be dialog?

**Comment by kball** [comment]
cc @Owlbertz

**Comment by Owlbertz** [comment]
Well I guess as Foundation aims at being the most advanded framework, we should adapt to the latest specification, therefore I think yes, you are correct and we should go with`aria-haspopup="dialog"` here.

**Comment by jaarong** [comment]
Thanks, but I wouldn't take my word for it.  I have literally made a change across the board to our theme one day and reverted the commit the next after reading new information.  

For instance, this [W3 dialog modal example page](http://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html) is using a button to open the window (which we have found is more screen reader accessible for this purpose than an anchor tag), and the button doesn't use aria-haspopup at all.  It also uses aria-modal="true" on the modal div which is also a part of the 1.1 standard.  

To add to the confusion, I found this thread about a [making the react-modal more accessible](https://github.com/reactjs/react-modal/issues/237) that says role="dialog" shouldn't be a definite value because of it's effect on screen readers, and should be optional which is they way they decided to go.  

All in all, I am confused and this reminds me of the golden days of the browser wars.  Our weakness is that we don't have all the screenreader and assistive technology versions to test with.  I would love a caniuse.com for accessibility.

I think I was more looking for conformation for what should be done in this specific instance rather than trying to provide guidance.

**Comment by Owlbertz** [comment]
Yeah, it's a shame that there are so many interpretations/implementations out there.
However, I still feel like we should stick to the (imo) most relevant standard which is the one you linked to.

What does @kball think?

**Comment by andycochran** [comment]
Since we're not sure that this is a bug, I'm going to close this issue. But let's make sure to consider this for v7, making sure that the docs clearly explain what's best practice and what's required with aria values.
